src/Nethermind/Nethermind.Blockchain/BaseLoopBlockProducer.cs

```diff
@@ -45,7 +49,10 @@ public abstract class BaseLoopBlockProducer : BaseBlockProducer
 
         public override void Start()
         {
-            _producerTask = Task.Run(ProducerLoop, CancellationTokenSource.Token).ContinueWith(t =>
+            Processor.ProcessingQueueEmpty += OnBlockProcessorQueueEmpty;
+            BlockTree.NewBestSuggestedBlock += BlockTreeOnNewBestSuggestedBlock;
```

Comment:
does it take into account both recovery and processing queues? if recovery queue is accounted for then this would work

Reply:
This is same as MinedBlockProducer and generally works ok

Reply:
Added recovery queue to the logic

Reply:
it would fail on very slow blocks, with rec queue it is fine